File: bbb-video/src/main/java/org/bigbluebutton/app/video/MeetingManager.java
package org.bigbluebutton.app.video;

import java.util.HashMap;
import java.util.Map;

public class MeetingManager {

    private VideoApplication app;

    private Map<String, Meeting> meetings = new HashMap<String, Meeting>();

    public MeetingManager(VideoApplication app) {
        this.app = app;
    }

    public void add(String id) {
        Meeting m = new Meeting(id);
        meetings.put(m.id, m);
    }

    public void remove(String id) {
        Meeting m = meetings.remove(id);
        if (m != null) {
            // Close all streams;
        }
    }

}

File: bbb-apps-common/src/main/scala/org/bigbluebutton/client/meeting/MeetingManager.scala
package org.bigbluebutton.client.meeting

object MeetingManager {
  def findWithMeetingId(manager: MeetingManager, meetingId: String): Option[Meeting] = {
    manager.toVector.find(m => m.meetingId == meetingId)
  }

  def remove(manager: MeetingManager, meetingId: String): Option[Meeting] = {
    manager.remove(meetingId)
  }

  def add(manager: MeetingManager, meeting: Meeting): Meeting = {
    manager.save(meeting)
  }

}

class MeetingManager {
  private var meetings = new collection.immutable.HashMap[String, Meeting]

  private def toVector: Vector[Meeting] = meetings.values.toVector

  private def save(meeting: Meeting): Meeting = {
    meetings += meeting.meetingId -> meeting
    meeting
  }

  private def remove(id: String): Option[Meeting] = {
    val meeting = meetings.get(id)
    meeting foreach (u => meetings -= id)
    meeting
  }
}
